ui/client/components/graph/node-modal/editors/expression/duration/PeriodEditor.tsx

```diff
@@ -23,23 +24,31 @@ type Props = {
 const SPEL_PERIOD_DECODE_REGEX = /^T\(java\.time\.Period\)\.parse\('(.*?)'\)$/
 const SPEL_PERIOD_SWITCHABLE_TO_REGEX = /^T\(java\.time\.Period\)\.parse\('P([0-9]{1,}Y)?([0-9]{1,}M)?([0-9]{1,}W)?([0-9]{1,}D)?'\)$/
 const SPEL_FORMATTED_PERIOD = (isoPeriod) => `T(java.time.Period).parse('${isoPeriod}')`
+const UNDEFINED_PERIOD = {
+  years: () => undefined,
+  months: () => undefined,
+  days: () => undefined
+}
 
 export default function PeriodEditor(props: Props) {
 
   const {expressionObj, onValueChange, validators, showValidation, readOnly, isMarked} = props
 
+  function isPeriodUndefined(period: Period): boolean {
+    return period.years === undefined && period.months === undefined && period.days === undefined
```

Comment:
I think that we should assume that if some part of period is defined then other values are implicitly zero and whole period is defined. Also wdyt about changing logic from negative to positive (isPeriodDefined)? The same in duration